addons/website/controllers/form.py

```diff
@@ -244,7 +245,7 @@ def insert_record(self, request, model, values, custom, meta=None):
             # If there isn't, put the custom data in a message instead
             if default_field.name:
                 if default_field.ttype == 'html' or model_name == 'mail.mail':
-                    custom_content = nl2br(custom_content)
+                    custom_content = nl2br(escape(custom_content))
                 record.update({default_field.name: custom_content})
             else:
                 values = {
```

Comment:
The `body` field of `mail.message` is also an HTML field - shouldn't it also be escaped before being given to `nl2br` ?